dot/network/service.go

```diff
@@ -304,6 +305,12 @@ func (s *Service) logPeerCount() {
 	}
 }
 
+type peerInfo struct {
+	Roles      byte   `json:"roles"`
+	BestHash   string `json:"bestHash"`
+	BestNumber uint64 `json:"bestNumber"`
+}
+
 func (s *Service) publishNetworkTelemetry(done chan interface{}) {
```

Comment:
wondering why this uses ` done chan` instead of the service `ctx`?

Reply:
I was following suggestions from this thread: https://github.com/ChainSafe/gossamer/pull/1528#discussion_r617684192